pkgs/by-name/ol/ollama/package.nix

```diff
@@ -133,11 +144,17 @@ goBuild {
     lib.optionalAttrs enableRocm {
       ROCM_PATH = rocmPath;
       CLBlast_DIR = "${clblast}/lib/cmake/CLBlast";
+      HIP_PATH = rocmPath;
     }
-    // lib.optionalAttrs enableCuda { CUDA_LIB_DIR = "${cudaToolkit}/lib"; };
+    // lib.optionalAttrs enableCuda {
+      CUDA_PATH = "${cudaPath}";
```

Comment:
it is because of this https://github.com/ollama/ollama/blob/4759d879f2376ffb9b82f296e442ec8ef137f27b/llama/make/Makefile.cuda_v12#L8

in the nix store we have for example `/nix/store/gdph9rnz9b46c73rm7d20hxkk3m394kq-cuda-merged-12.2/ ` which will not match.

Reply:
Yes, the version information is currently hardcoded in the Makefile.

Regarding `CUDA_PATH`, it's still necessary. Although you can successfully build `ollama-cuda` without explicitly setting it, you'll encounter missing runner issues at runtime. refer to the log in https://github.com/ollama/ollama/issues/7597#issuecomment-2466762267